Framework/Core/include/Framework/DataRefUtils.h

```diff
@@ -28,13 +28,13 @@ struct DataRefUtils {
   // a POD, this is to distinguish it from the alternative below,
   // which works for TObject (which are serialised).
   template <typename T>
-  static typename std::enable_if<std::is_pod<T>::value == true, Collection<T>>::type
+  static typename std::enable_if<std::is_pod<T>::value == true, gsl::span<T>>::type
   as(DataRef const &ref) {
     using DataHeader = o2::Header::DataHeader;
     auto header = o2::Header::get<const DataHeader>(ref.header);
     assert((header->payloadSize % sizeof(T)) == 0);
```

Comment:
probably we should also check the serialization type to be `gSerializationMethodNone`
throw `runtime_error` exception? The `assert` will be gone when compiling with `NDEBUG`

Reply:
Yeah, I guess so, although I am not sure `DataRefUtils` should be the one which does checks and throws exceptions rather than the upper layer. Will update it for now.